include/nbl/video/IGPUFence.h

```diff
@@ -31,6 +33,94 @@ class IGPUFence : public core::IReferenceCounted, public IBackendObject
     virtual ~IGPUFence() = default;
 };
 
+class GPUEventWrapper : public core::Uncopyable
+{
+protected:
+    ILogicalDevice* mDevice;
+    core::smart_refctd_ptr<IGPUFence> mFence;
```

Comment:
doesnt the fence hold a pointer to the logical device anyway?

Reply:
hm, yeah it does, a const one though. But, yeah

Reply:
and you need a non-const?